sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3997,6 +3997,10 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         c.markAsAnalyzed(AnalysisContext.get)
       case c: KeepAnalyzedQuery if c.resolved =>
         c.storeAnalyzedQuery()
+      case p: LogicalPlan if p.resolved =>
```

Comment:
I can, in theory, limit it to commands, as all current use cases belong to commands, but not sure it is a good idea.

Reply:
This is confusing, both the trait name `AnalysisAwareExpression` and the method `markAsAnalyzed`. When the child expression is resolved, the `DefaultValueExpression` is consider resolved. Why do we need this?

Reply:
nvm, I think `AnalysisAwareExpression` is a fine name, as what Spark does is to notify it when it's fully analyzed.